at::launch benchmark

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21582 Parallelize bi-LSTM implementation
* **#21581 at::launch benchmark**
* #20480 Native TBB parallel backend

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

Differential Revision: [D15736495](https://our.internmc.facebook.com/intern/diff/D15736495)

Commit:
at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

cpuhrsch commented on binaries/at_launch_benchmark.cc:
```diff
@@ -0,0 +1,81 @@
+    at::set_num_interop_threads(FLAGS_inter_op_threads);
+  }
+
+  launch_tasks_and_wait(FLAGS_warmup_iter);
```
As with the benchmark in https://github.com/pytorch/pytorch/pull/20480 I'd explicitly time this to see if there's an initialization overhead and run the main benchmark multiple times to see if the timing is consistent across runs. In general I'd run the benchmark until a certain amount of time has passed and not until it has been executed a certain number of times.

ilia-cher replied:
I'll add the timing information

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

ilia-cher commented:
updated with extra timing information

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head

Commit:
Update on "at::launch benchmark"

at::launch benchmark

Summary:
Simple at::launch benchmark

Test Plan:
USE_CUDA=0 BLAS=MKL USE_MKLDNN=1 BUILD_BINARY=1 python setup.py develop
--cmake
./build/bin/at_launch_benchmark

gh-metadata: pytorch pytorch 21581 gh/ilia-cher/26/head
